src/topology/homotopy/H_spaces.lean

```diff
@@ -0,0 +1,150 @@
+
+lemma Q_right_one_left (θ : I) : Q_right (1, θ) = 1 :=
+set.proj_Icc_of_right_le _ $ (le_div_iff $ add_pos zero_lt_one).2 $
+  by { dsimp only, rw [coe_one, one_mul, mul_one], apply add_le_add_left (le_one _) }
+
+lemma Q_right_zero_right (t : I) : (Q_right (t, 0) : ℝ) = if (t : ℝ) ≤ 1 / 2 then 2 * t else 1 :=
+begin
+  simp only [Q_right, coe_zero, add_zero, div_one],
+  split_ifs,
+  { rw set.proj_Icc_of_mem _ ((mul_pos_mem_iff zero_lt_two).2 _), exacts [rfl, ⟨t.2.1, h⟩] },
+  { rw (set.proj_Icc_eq_right _).2, { refl }, { linarith }, { exact zero_lt_one } },
+end
+
+lemma Q_right_one_right (t : I) : Q_right (t, 1) = t :=
+eq.trans (by {rw Q_right, congr, apply mul_div_cancel_left, exact two_ne_zero}) $
+  set.proj_Icc_coe zero_le_one _
+
+end unit_interval
+
+namespace path
+
+open unit_interval
+
+variables {X : Type u} [topological_space X] {x y : X}
+
+/-- This is the function analogous to the one on p. 475 of [serre1951], defining a homotopy from
+the product path `e ∧ γ` to `γ`.-/
+def delayed_refl_right (θ : I) (γ : path x y) : path x y :=
+{ to_fun := λ t, γ (Q_right (t, θ)),
+  continuous_to_fun := γ.continuous.comp (continuous_Q_right.comp $ continuous.prod.mk_left θ),
+  source' := by { dsimp only, rw [Q_right_zero_left, γ.source] },
+  target' := by { dsimp only, rw [Q_right_one_left, γ.target] } }
+
+lemma continuous_delayed_refl_right : continuous (λ p : I × path x y, delayed_refl_right p.1 p.2) :=
+continuous_uncurry_iff.mp $ (continuous_snd.comp continuous_fst).path_eval $
+  continuous_Q_right.comp $ continuous_snd.prod_mk $ continuous_fst.comp continuous_fst
+
+lemma delayed_refl_right_zero (γ : path x y) : delayed_refl_right 0 γ = γ.trans (path.refl y) :=
+begin
+  ext t,
+  simp only [delayed_refl_right,
+    trans_apply, refl_extend, path.coe_mk, function.comp_app, refl_apply],
+  split_ifs, swap, conv_rhs { rw ← γ.target },
+  all_goals { apply congr_arg γ, ext1, rw Q_right_zero_right },
+  exacts [if_neg h, if_pos h],
+end
+
+lemma delayed_refl_right_one (γ : path x y) : delayed_refl_right 1 γ = γ :=
+by { ext t, exact congr_arg γ (Q_right_one_right t) }
+
+/-- This is the function on p. 475 of [serre1951]], defining a homotopy from a path `γ` to the
+product path `e ∧ γ`.-/
+def delayed_refl_left (θ : I) (γ : path x y) : path x y := (delayed_refl_right θ γ.symm).symm
+
+lemma continuous_delayed_refl_left : continuous (λ p : I × path x y, delayed_refl_left p.1 p.2) :=
+path.continuous_symm.comp $ continuous_delayed_refl_right.comp $ continuous_fst.prod_mk $
+  path.continuous_symm.comp continuous_snd
+
+lemma delayed_refl_left_zero (γ : path x y) : delayed_refl_left 0 γ = (path.refl x).trans γ :=
+by simp only [delayed_refl_left, delayed_refl_right_zero, trans_symm, refl_symm, path.symm_symm]
+
+lemma delayed_refl_left_one (γ : path x y) : delayed_refl_left 1 γ = γ :=
+by simp only [delayed_refl_left, delayed_refl_right_one, path.symm_symm]
+
+notation ` Ω_[` x `]` := path x x
```

Comment:
If this notation should be global, it should be defined near `path`.

Reply:
You mean in `homotopy/path.lean`? I can move it there.

Reply:
Yes, that would be better. (Also: I think the printing of the notation might be nicer if you remove the space before the `Ω`)

Reply:
OK, I will move it there and remove the space.

Reply:
I ended up removing the notation, since `homotopy/path`  has already been ported (and the notation was used only in the final lemma).